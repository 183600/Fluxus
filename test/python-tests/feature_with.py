# Context manager tests

class FileManager:
    def __init__(self, filename):
        self.filename = filename
        self.file = None
    
    def __enter__(self):
        print(f"Opening file: {self.filename}")
        # Simulate opening a file
        self.file = f"file_handle_{self.filename}"
        return self.file
    
    def __exit__(self, exc_type, exc_val, exc_tb):
        print(f"Closing file: {self.filename}")
        if exc_type:
            print(f"Exception occurred: {exc_type.__name__}")
        return False

# Test with statement
with FileManager("test.txt") as file:
    print(f"Working with {file}")

print("File operation completed")