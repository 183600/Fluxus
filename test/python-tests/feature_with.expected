Opening file: test.txt
Working with file_handle_test.txt
Closing file: test.txt
File operation completed